project1/application.py

```diff
@@ -26,31 +29,97 @@
+        return render_template("index.html", flag = True, name = session.get("name"))
     return render_template("index.html", flag = False)
 
 @app.route("/register", methods = ["GET", "POST"])
 def register():  
-    if request.method == "POST":       
-        db.session.add(User(request.form.get("name"), request.form.get("password")))
-        return render_template("register.html", flag = True, name = request.form.get("name"))
-    return render_template("register.html", flag = False)
+
+    # If the user fills the form and submits it.
+    if request.method == "POST":
+
+        # Checking whether all the fields are filled.
+        if not request.form.get("name"):
+            return render_template("error.html", message="Please provide User name.")
+        if not request.form.get("password"):
+            return render_template("error.html", message="Please provide Password.")
+
+        # Trying to see whether the user is already present or not. If present
+        # will ask the user to login.
+        try:
+            existing = User.query.filter_by(name=request.form.get("name")).one()
+            return render_template("register.html", flag = True,
+                name = request.form.get("name"),
+                message = """It seems like you are already a
+                 registered user with us. Please use login button next time.""")
+
+        # If the user is not added to our we will add that person to the
+        # database.
+        except:
+            try:  
+                db.session.add(User(request.form.get("name"),
+                 request.form.get("password")))
+                db.session.commit()
+
+                os.system("flask db init")
+                os.system("flask db migrate")
+                os.system("flask db upgrade")
+
+                return render_template("register.html", flag = True,
+                        name = request.form.get("name"),
+                        message = """Aww yeah, you successfully registered
+                         for this application.""")
+            except:
+                return render_template("register.html", flag = False,
+                    message = """There was some error on our side. 
+                    Please try registering again.""")
+    # This is the for get request to present the register page.
+    return render_template("register.html", flag = False, message = "")
 
 @app.route("/admin")
 def admin():
+    # This method shows all the users in a table format.
     users = User.query.order_by(User.timestamp.desc()).all()
     return render_template("admin.html", users = users)
+
+@app.route("/auth", methods = ["POST"])
+def auth():
+
+    # Tries to see whether the user provided the details correct or not.
+    # Otherwise will redirect to login page.
+    if request.method == "POST":
+        try:
+            session['name'] = request.form.get("name")
+            session['password'] = request.form.get("password")
```

Comment:
I think it is enough if we add name to session, we need not add password

Reply:
> I think it is enough if we add name to session, we need not add password

Resolved the issue.